provider/src/main/kotlin/au/com/dius/pact/provider/ProviderVerifier.kt

```diff
@@ -844,6 +848,30 @@ open class ProviderVerifier @JvmOverloads constructor (
     }
   }
 
+  private fun reportResults(
+      pact: FilteredPact,
+      result: VerificationResult,
+      client: IPactBrokerClient?
+  ): Result<Boolean, List<String>> {
+    if (providerBranch?.get()?.isNotBlank() == true){
```

Comment:
This seems to only allow publishing with tags OR a branch. It should allow both (or neither) to be used. The branch should be created before the tags.

Reply:
Thanks for explanation. I don't know why I thought it should be either "tags" or "branch", the specification says "before creating the tags, send a PUT using the pb:branch-version" :)
I will add PUT request to the branch endpoint just before the call to the tags endpoint and remove the conditional logic.